### PR TITLE
CUMULUS-1635 add CMA metadata to log messages

### PR DESCRIPTION
**Pull request by chuckwondo**
Fixes CUMULUS-1635

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
+[![CircleCI]](https://circleci.com/gh/nasa/cumulus-message-adapter-python)
+[![PyPI version]](https://badge.fury.io/py/cumulus-message-adapter-python)
 
 ## What is Cumulus?
 
 Cumulus is a cloud-based data ingest, archive, distribution and management
 prototype for NASA's future Earth science data streams.
 
-Read the [Cumulus Documentation](https://cumulus-nasa.github.io/)
+Read the [Cumulus Documentation]
```
Comment (markdboyd) [inline README.md]: Seems like you unintentionally removed the link here

Comment (chuckwondo) [reply README.md]: I moved all links to the bottom of the md file.

File: README.md [hunk 2]
```diff
@@ -217,3 +230,14 @@ This approach has a few major advantages:
 ## License
 
 [Apache 2.0](LICENSE)
+
+[circleci]:
+  https://circleci.com/gh/nasa/cumulus-message-adapter-python.svg?style=svg
+[pypi version]:
+  https://badge.fury.io/py/cumulus-message-adapter-python.svg
+[Cumulus Documentation]:
+  https://cumulus-nasa.github.io/
```
Comment (markdboyd) [inline README.md]: This isn't your error, but this link is wrong. It should be `https://nasa.github.io/cumulus/`

Comment (chuckwondo) [reply README.md]: Thanks. Fixed.

**Comment by jennyhliu** [comment]
I tested the example task in aws, and the log entries in CloudWatch for this python is different from what I see in js (you can locate one in `MoveGranules` log) or java.
js:
```json
{
    "executions": "6cea823b-b8fe-4dfc-bd64-ca560325b4c1",
    "granules": "[\"MOD09GQ.A0769358.PgkxGU.006.7342153536624\"]",
    "level": "debug",
    "message": "Configured duplicateHandling value: replace, forceDuplicateOverwrite false",
    "sender": "jl-tf-MoveGranules",
    "stackName": "jl-tf",
    "timestamp": "2019-11-26T19:26:05.193Z",
    "version": "$LATEST"
}
```
python:
```json
{
    "sender": "jl-tf-testpython",
    "level": "error",
    "timestamp": "2019-11-27T19:22:12.186728",
    "executions": [
        "execution_value"
    ],
    "stackName": null,
    "version": "$LATEST",
    "parentArn": null,
    "asyncOperationId": null,
    "message": "task formatted message bar exc_info  None",
    "granules": [
        "MOD09GQ.A2016358.h13v04.006.2016360104606",
        "MOD09GQ.A2016358.h13v04.007.2017"
    ]
}
```
1. `executions` is a string with single execution name
2. `granules` stringified json string with list of granuleIds.
3. Also, in js/java implementation, if a field is not availlable, such as asyncOperationId, it won't show up in the log rather than field with `null` value.

**Comment by jennyhliu** [comment]
Also, suggest to add tests with the sample messages from and verify the log messages have correct contents.
https://github.com/nasa/cumulus-message-adapter-js/tree/master/test/fixtures/messages

File: cumulus_logger.py [hunk 3]
```diff
@@ -1,99 +1,315 @@
...
+        Passes the supplied arguments to ``createMessage`` and logs the
+        returned message dict.  See ``createMessage``.
+        """
+        self._log(logging.CRITICAL, message, *args, **kwargs)
+
+    def _resolve_log_level(self, level_name):
+        if not isinstance(level_name, str):
+            return self.logger.level
+        level = logging.getLevelName(level_name.upper())
+        return level if isinstance(level, (int, long)) else self.logger.level
```
Comment (jennyhliu) [inline cumulus_logger.py]: We need to support both python27 and python36, this works for Python 2 only.
Use one of these:
```python
from builtins import int    # subclass of long on Py2
if isinstance(x, int)

from past.builtins import long
if isinstance(x, (int, long))
```

Comment (chuckwondo) [reply cumulus_logger.py]: I ran this locally in Python 2.7 and the CI tests for 2.7 also passed. Did you see this fails somehow?

**Comment by chuckwondo** [comment]
> I tested the example task in aws, and the log entries in CloudWatch for this python is different from what I see in js (you can locate one in `MoveGranules` log) or java.
> js:
> 
> ```json
> {
>     "executions": "6cea823b-b8fe-4dfc-bd64-ca560325b4c1",
>     "granules": "[\"MOD09GQ.A0769358.PgkxGU.006.7342153536624\"]",
>     "level": "debug",
>     "message": "Configured duplicateHandling value: replace, forceDuplicateOverwrite false",
>     "sender": "jl-tf-MoveGranules",
>     "stackName": "jl-tf",
>     "timestamp": "2019-11-26T19:26:05.193Z",
>     "version": "$LATEST"
> }
> ```
> 
> java:
> 
> ```json
> {
>     "executions": "execution_value",
>     "sender": "jlTestTaskJava",
>     "level": "info",
>     "granules": "[\"MOD09GQ.A2016358.h13v04.006.2016360104606\",\"MOD09GQ.A2016358.h13v04.007.2017\"]",
>     "message": "Business logic input: {\"input\": {\"anykey\": \"anyvalue\"}, \"config\": {\"bar\": \"baz\"}}",
>     "version": "$LATEST",
>     "timestamp": "2019-11-15 22:27:04.379"
> }
> ```
> 
> python:
> 
> ```json
> {
>     "sender": "jl-tf-testpython",
>     "level": "error",
>     "timestamp": "2019-11-27T19:22:12.186728",
>     "executions": [
>         "execution_value"
>     ],
>     "stackName": null,
>     "version": "$LATEST",
>     "parentArn": null,
>     "asyncOperationId": null,
>     "message": "task formatted message bar exc_info  None",
>     "granules": [
>         "MOD09GQ.A2016358.h13v04.006.2016360104606",
>         "MOD09GQ.A2016358.h13v04.007.2017"
>     ]
> }
> ```
> 
> ```json
> {
>     "sender": "jl-tf-testpython",
>     "level": "info",
>     "timestamp": "2019-11-27T19:47:42.392491",
>     "executions": [
>         "execution_value"
>     ],
>     "stackName": "cumulus-stack",
>     "version": "$LATEST",
>     "parentArn": "arn:aws:states:us-east-1:12345:execution:DiscoverGranules:8768aebb",
>     "asyncOperationId": "async-id-123",
>     "message": "task executed",
>     "granules": []
> }
> ```
> 
> 1. `executions` is a string with single execution name
> 2. `granules` is a stringified json string with list of granuleIds.
> 3. Also, in js/java implementation, if a field is not available or empty, such as asyncOperationId, it won't show up in the log rather than field with `null` value.  The python lib has `null` or `[]` field values.

Ok, just to be sure I understand the changes requested, are the non-null values showing up correctly in the python output, even if the format is not what you expect? I'm asking because the values you show in the python output are different from the values you show in the js and java outputs. I want to make sure the python values are correct, even if they are not formatted the same.

**Comment by chuckwondo** [comment]
> Also, suggest to add tests with the sample messages from and verify the log messages have correct contents.
> https://github.com/nasa/cumulus-message-adapter-js/tree/master/test/fixtures/messages

Ok, I'll take a look at that.

File: cumulus_logger.py (hunk 3 above)
Comment (jennyhliu) [reply cumulus_logger.py]: It causes error in python 3.6.
pylint for python3 has: Undefined variable: 'long'
lambda with python 3.6 has: name 'long' is not defined: NameError

**Comment by jennyhliu** [comment]
https://github.com/nasa/cumulus-message-adapter-python/pull/32#issuecomment-559275142
The CW log entries are from different tasks (different events), so they have different field values.  What I tried to point out is that, the python logger should have same log format as other client libraries(js, java), so that it can be parsed the same way.  The python output seems to have the correct values.

File: cumulus_logger.py (hunk 3 above)
Comment (jennyhliu) [reply cumulus_logger.py]: I don't know why the tests passed in python3.6, that's something we want to check.

File: run_cumulus_task.py [hunk 4]
```diff
@@ -45,20 +51,21 @@ def run_cumulus_task(
     context_dict = vars(context) if context else {}
     logger = CumulusLogger()
     logger.setMetadata(cumulus_message, context)
-    message_adapter_disabled = os.environ.get('CUMULUS_MESSAGE_ADAPTER_DISABLED')
+    message_adapter_disabled = os.environ.get(
+        'CUMULUS_MESSAGE_ADAPTER_DISABLED', '').lower()
 
-    if message_adapter_disabled is 'true':
+    if message_adapter_disabled in ['true', 't', 'yes', 'y', 'on', '1']:
```
Comment (jennyhliu) [inline run_cumulus_task.py]: To be consistent with other CMA client library, only 'true' value should be used.

**Comment by jennyhliu** [comment]
Need a new release.

File: run_cumulus_task.py (hunk 4 above)
Comment (chuckwondo) [reply run_cumulus_task.py]: Will do.

File: cumulus_logger.py (hunk 3 above)
Comment (chuckwondo) [reply cumulus_logger.py]: Yeah, odd. I created a Python 3.6 env to run the tests and I'm able to reproduce the error. I forgot that Python 3 dropped `long`. I'll make the correction.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -35,10 +47,25 @@ All of the automated tests for this project need to pass before your submission
...
 ```
 
-### For more information on Cumulus governance, see the [Cumulus Code Contribution Guidelines](https://docs.google.com/document/d/14J_DS6nyQ32BpeVjdR-YKfzHAzFB299tKghPGshXUTU/edit) and [the Cumulus Wiki](https://wiki.earthdata.nasa.gov/display/CUMULUS/Cumulus).
+**For more information on Cumulus governance, see the [Cumulus Code Contribution
+Guidelines] and the [Cumulus Wiki]**
+
+[search open issues]:
+  https://github.com/cumulus-nasa/cumulus-message-adapter-python/issues
+[create a new issue]:
+  https://github.com/cumulus-nasa/cumulus-message-adapter-python/issues/new
```
Comment (jennyhliu) [inline CONTRIBUTING.md]: Please also update this file with new URL.
https://github.com/nasa/cumulus-message-adapter-python/blob/master/example/requirements.txt

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -35,10 +47,25 @@ All of the automated tests for this project need to pass before your submission
...
+Improve contributing docs and consolidate them in the standard location
+https://help.github.com/articles/setting-guidelines-for-repository-contributors/
 ```
 
-### For more information on Cumulus governance, see the [Cumulus Code Contribution Guidelines](https://docs.google.com/document/d/14J_DS6nyQ32BpeVjdR-YKfzHAzFB299tKghPGshXUTU/edit) and [the Cumulus Wiki](https://wiki.earthdata.nasa.gov/display/CUMULUS/Cumulus).
+**For more information on Cumulus governance, see the [Cumulus Code Contribution
+Guidelines] and the [Cumulus Wiki]**
+
+[search open issues]:
+  https://github.com/cumulus-nasa/cumulus-message-adapter-python/issues
```
Comment (jennyhliu) [inline CONTRIBUTING.md]:
```suggestion
  https://github.com/nasa/cumulus-message-adapter-python/issues
```

File: CONTRIBUTING.md (hunk 5 above)
Comment (jennyhliu) [inline CONTRIBUTING.md]:
```suggestion
  https://github.com/nasa/cumulus-message-adapter-python/issues/new
```

**Pull request merged**